sympy/solvers/bivariate.py

```diff
@@ -169,15 +170,16 @@ def _lambert(eq, x):
     # solution of the given Lambert equation is like
     # sol = -c/b + (a/d)*LambertW(arg, k),
     # where arg = d/(a*b)*exp((c*d-b*f)/a/b) and k in lambert_real_branches.
-    # arg can have values of p roots of exp(num/den)**(1/p)[exp((c*d-b*f)/a/b)]
-    # and will required to have all solutions of x**p - exp(num/den).
+    # arg can have values of p roots of exp((c*d-b*f)/a/b) represented
+    # as exp(num/den)**(1/p) will required to have all
```

Comment:
Is it okay @smichr ?

Reply:
Maybe
```
Instead of considering the single arg, `d/(a*b)*exp((c*d-b*f)/a/b)`, the individual `p` roots obtained when writing `exp((c*d-b*f)/a/b)` as `exp(A/p) = exp(A)**(1/p)`, where `p` is an Integer, are used.
```

Are you able to trace back (like you did with the need to consider even powers) to why this needs to be done for `exp`'s argument?

Reply:
> Instead of considering the single arg, `d/(a*b)*exp((c*d-b*f)/a/b)`, the individual `p` roots obtained when writing `exp((c*d-b*f)/a/b)` as `exp(A/p) = exp(A)**(1/p)`, where `p` is an Integer, are used.

okay.
> Are you able to trace back (like you did with the need to consider even powers) to why this needs to be done for exp's argument?

The trace back is easy to understand. Please try to read between lines if i miss some thing.
Earlier the function`_lambert` has the working like :- 
1. Find all the values of a, b, c, d, e in the required loagrithmic equation
2. Then it defines a solution of the form 
`-c/b + (a/d)*l where l = LambertW(d/(a*b)*exp(c*d/a/b)*exp(-f/a), k)`
and then it included that solution.

I agree everything seems flawless here. but try to see the step where we are defining l.
Let us suppose a hypothetical algorithm just like algorithm used in `_lambert` in which equation to be solved is
`x**3 - 1 = 0` and in which we define solution of the form 
`x = exp(I*2*pi/n) where n is the power of x in equation`
so the algorithm will give solution `x = exp(I*2*pi/3) but expected was [1, exp(I*2*pi/3), exp(-I*2*pi/3)]` which can be found by finding all solutions of `x**n - exp(2*I*pi) = 0` by a different correct algorithm. Thats y it was wrong.
The above algorithm would have given correct values for ` x - 1= 0 ` 

And the question in your mind may arise that y only exp() because the possiblity of having more than one roots is in exp(), because if the algorithm would have been like ` x = a, where a is some real constant then there is not any possiblity of further roots rather than solution like x = a**(1/n)`

Reply:
Please give your confirmation if you agreed to this and I had updated the comment.